### ISSUE TITLE
Switch payment processor payment type to 'direct debit'

### ISSUE THREAD

**Issue by systopia** [Project60/org.project60.sepa#249]
Due to the lack of proper documentation, the SDD payment processor has so far been using the 'credit' payment type. We now want to switch this to direct debit, to comply with the "standard" payment process.

**Comment by systopia** [comment]
Done with https://github.com/Project60/org.project60.sepa/commit/9cb0f063b62a8a76e5e11230c7a017fccd1704de. 

This is a rather sensitive change to the payment process, so the module should be regression tested.

In addition, already existing payment processors need to be updated:
```
UPDATE civicrm_payment_processor SET payment_type = 2 WHERE class_name = 'Payment_SDD';
```

**Comment by tttp** [comment]
Hi,

As far as I remember, I tried first the payment_type=2 on the PP, but had an issue (beside being forced to use field names that quite don't march ;) down the road...

And obviously don't remember what. is this that other fields (eg billing address) was there? or that some messages where hardcoded and didn't make any sense in the case of a sepa? Not sure and might not be an issue anymore on 4.5, but as you wrote, rather sensitive change, so risk of regression.

For what it's worth, I had to tweak the billing-block that way on the initial payment processor, might be linked:
    CRM_Core_Region::instance('billing-block')->update('default', array('disabled' => TRUE));
    CRM_Core_Region::instance('billing-block')->add(array('template' => 'CRM/Sepa/Mandate.tpl',
          'weight' => -1));

**Comment by systopia** [comment]
@nikobo: Could you please check if something like this still affects us? To me the process now sees fine on 4.4.11.

**Comment by antrik** [comment]
The original PP used an "unofficial" Payment Type of 9000, which is relevant to ISO 20022 I guess, but has no meaning for CiviCRM. That didn't seem to matter though -- we only had to switch it to 1 ('Credit Card') in our branch when we added SDD support in the backoffice form, which ignores any processors not marked as 'Credit Card'.

This didn't cause any side effects in other places as far as I know. For the most part, it shouldn't really matter, as we switch out all the billing fields anyways -- though the fact that we switch out *other* fields might cause side effects in theory. I know there are *some* code paths in CiviCRM that distinguish the types; so unexpected differences cannot be ruled out entirely -- especially as the 'Debit Card' code paths are very incomplete and poorly tested in general. (Mostly the backend parts, which are clearly unfinished and buggy in obvious ways; but IIRC the frontend is somewhat flaky as well, with at least one issue on file...)

All in all, I consider it a minefield, with no obvious benefits, and only possible risks.

Also note that Payment Type 2 is actually 'Debit Card'. This is *not* what we are doing here. SDD doesn't fit *either* of the existing types: they are both inappropriate. Introducing a separate type (like the original PP did) is indeed the most correct approach -- and adding some tweaks in CiviCRM core so it doesn't treat everything as card payments, this could indeed reduce the amount of hacks necessary...
